fix(bridge)_: fix bridge replies not working when they come from Discord

jrainville opened the pull request:
Fixes https://github.com/status-im/status-desktop/issues/16323

The problem was that the code expected to receive the Discord message ID, but the bridge is smart enough to return the Status message ID already, so there is no need to try and convert it.

There were also a couple issues in the bridge code itself.

Bridge PR: https://github.com/status-im/matterbridge/pull/18

jrainville committed:
fix(bridge)_: fix bridge replies not working when they come from Discord

Fixes https://github.com/status-im/status-desktop/issues/16323

The problem was that the code expected to receive the Discord message ID, but the bridge is smart enough to return the Status message ID already, so there is no need to try and convert it.

There were also a couple issues in the bridge code itself.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (20)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | 1c5288b7 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/1/) | 2024-09-12 22:06:46 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/1//consoleText) |
| :heavy_check_mark: | 1c5288b7 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/1/) | 2024-09-12 22:07:45 | ~2 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/1//consoleText) |
| :heavy_check_mark: | 1c5288b7 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5830/1/) | 2024-09-12 22:09:23 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-240912-220525-1c5288-pr5830.zip) |
| :heavy_check_mark: | 1c5288b7 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5830/1/) | 2024-09-12 22:10:24 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240912-220523-1c5288-pr5830.zip) |
| :heavy_check_mark: | 1c5288b7 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5830/1/) | 2024-09-12 22:10:52 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240912-220522-1c5288-pr5830.aar) |
| | | | | | | |
| :heavy_check_mark: | d93e6a98 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5830/2/) | 2024-09-12 22:12:46 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240912-221115-d93e6a-pr5830.aar) |
| :heavy_check_mark: | d93e6a98 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/2/) | 2024-09-12 22:12:54 | ~2 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/2//consoleText) |
| :heavy_multiplication_x: | d93e6a98 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/2/) | 2024-09-12 22:12:54 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/2//consoleText) |
| :heavy_check_mark: | d93e6a98 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5830/2/) | 2024-09-12 22:14:00 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240912-221040-d93e6a-pr5830.zip) |
| :heavy_check_mark: | d93e6a98 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5830/2/) | 2024-09-12 22:14:07 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-240912-221030-d93e6a-pr5830.zip) |
| | | | | | | |
| :heavy_check_mark: | 2552a78b | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5830/3/) | 2024-09-13 13:56:59 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240913-135457-2552a7-pr5830.aar) |
| :heavy_check_mark: | 2552a78b | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5830/3/) | 2024-09-13 13:57:09 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-240913-135500-2552a7-pr5830.zip) |
| :heavy_check_mark: | 2552a78b | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/3/) | 2024-09-13 13:57:10 | ~2 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/3//consoleText) |
| :heavy_check_mark: | 2552a78b | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5830/3/) | 2024-09-13 13:58:43 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240913-135458-2552a7-pr5830.zip) |
| :heavy_multiplication_x: | 2552a78b | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/3/) | 2024-09-13 14:27:20 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | ba8236f0 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/4/) | 2024-09-23 17:09:27 | ~2 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/4//consoleText) |
| :heavy_multiplication_x: | ba8236f0 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/4/) | 2024-09-23 17:09:37 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/4//consoleText) |
| :heavy_check_mark: | ba8236f0 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5830/4/) | 2024-09-23 17:10:26 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-240923-170650-ba8236-pr5830.zip) |
| :heavy_check_mark: | ba8236f0 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5830/4/) | 2024-09-23 17:11:22 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240923-170646-ba8236-pr5830.aar) |
| :heavy_check_mark: | ba8236f0 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5830/4/) | 2024-09-23 17:12:40 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240923-170648-ba8236-pr5830.zip) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | c9023c00 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/5/) | 2024-09-24 18:17:42 | ~2 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/5//consoleText) |
| :heavy_check_mark: | c9023c00 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5830/5/) | 2024-09-24 18:18:55 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-240924-181452-c9023c-pr5830.zip) |
| :heavy_check_mark: | c9023c00 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5830/5/) | 2024-09-24 18:19:26 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240924-181450-c9023c-pr5830.zip) |
| :heavy_check_mark: | c9023c00 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5830/5/) | 2024-09-24 18:19:50 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240924-181448-c9023c-pr5830.aar) |
| :heavy_check_mark: | c9023c00 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/5/) | 2024-09-24 18:46:35 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 860e758a | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5830/6/) | 2024-09-27 14:58:20 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240927-145606-860e75-pr5830.aar) |
| :heavy_check_mark: | 860e758a | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5830/6/) | 2024-09-27 14:58:52 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-240927-145610-860e75-pr5830.zip) |
| :heavy_check_mark: | 860e758a | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/6/) | 2024-09-27 14:59:01 | ~2 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5830/6//consoleText) |
| :heavy_check_mark: | 860e758a | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5830/6/) | 2024-09-27 15:03:29 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240927-145607-860e75-pr5830.zip) |
| :heavy_check_mark: | 860e758a | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/6/) | 2024-09-27 15:28:09 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5830/6//consoleText) |

siddarthkay commented on protocol/message_persistence.go:
```diff
@@ -3075,15 +3075,7 @@ func (db sqlitePersistence) findAndUpdateReplies(tx *sql.Tx, bridgeMessageID str
 
 func (db sqlitePersistence) findAndUpdateRepliedTo(tx *sql.Tx, discordParentMessageID string, statusMessageID string) error {
-	// Finds status messages id which are replies for bridgeMessageID
-	repliedMessageID, err := db.findStatusMessageIDForBridgeMessageID(tx, discordParentMessageID)
```
nice! we could also nuke this function definition then `findStatusMessageIDForBridgeMessageID `

igor-sirotin commented:
@jrainville a bunch of tests are failing, including `TestBridgeMessageReplies` 🤔 
<img width="1121" alt="image" src="https://github.com/user-attachments/assets/b20c9a80-cfe6-4ec9-a8c3-5f77bd4f8870">

<img width="1028" alt="image" src="https://github.com/user-attachments/assets/756868af-15c4-4f6a-9075-b6471224bc24">

I think it's only `TestBridgeMessageReplies` and the others are just flaky.

igor-sirotin reviewed (approved):
Nice and clean, just fix the `TestBridgeMessageReplies`

jrainville committed:
fix_: fix broken test by even more simplification

jrainville commented:
@igor-sirotin @osmaczko re-asking review because I removed even more code when I fixed the test.

I also tested again with the real app to make sure my assumptions were all right and it still all works. Proof:
![image](https://github.com/user-attachments/assets/9cff31a2-52ad-4878-a5af-c7f49cbd73db)

Also, I tested closing the app and sending a lot of messages and replies, so that it would fetch out of order to confirm that it still managed to populate the replies even out of order and it worked:
![image](https://github.com/user-attachments/assets/3eb8582d-9206-422e-8984-3e8d4efe4013)

For example, the `4 4` message came in, but not the `4` message, so at first it shower "unknown original message, please fetch", I did fetch more and `4` appeared and it populated the reply correctly :+1:

igor-sirotin commented on protocol/persistence_test.go:
```diff
@@ -1995,18 +1995,18 @@ func TestBridgeMessageReplies(t *testing.T) {
 	require.NoError(t, err)
 
-	// status message "222" should have reply_to = "111"
+	// status message "222" should have reply_to =" 1" because it's a discord message to another discord message
```
These message ids are so unobvious, impossible to read the test really 😵‍💫
